src/components/input_field.tsx

```diff
@@ -42,152 +47,41 @@ export interface InputFieldProps {
     imageSize?: string;
     /** Shape of the image for type image: 'round' | 'square' */
     imageShape?: ImageShape;
-    /** If true, the defaultValue is shown as plain text and the input hidden */
-    staticField?: boolean
-    /** A list of suggestions for tags input field */
-    tagSuggestions?: string[]
-    /** Displays the required icon when true */
-    required?: boolean
-    /** Whether the form has been submitted. This will be set by the Form component */
-    submitted?: boolean
-    /** Input field label */
-    label?: string | JSX.Element;
-    /** Input field label width: '1/1' | '1/2' | '1/3' | '1/4' | '1/5' */
-    labelLayoutWidth?: LayoutWidth;
-    /** Input field label breakpoint: 'xs' | 'sm' | 'md' | 'lg' */
-    labelWidthBreakpoint?: Breakpoint;
-    /** JSX Element, such as an icon, to be shown before the input label */
-    preLabelElement?: JSX.Element
-    /** Description of the fields to be shown in a tooltip */
-    infoText?: string
-    /** The URL of the link to be shown next to the input field */
-    infoLinkURL?: string
-    /** Whether the textarea should grow horizontally with user input */
-    textareaAutoGrow?: boolean;
-}
-
-export interface InputFieldState {
-    /** Indicates if the field has been touched (changed) or not from the default value. */
-    touched: boolean;
 }
 
-export class InputField extends React.Component<InputFieldProps, InputFieldState> {
-
-    public static defaultProps: Partial<InputFieldProps> = {
+/**
+ * Class that adds an input wrapper around a ImageUpload component.
+ * TODO: merge the two components
+ */
+export class InputField extends React.Component<OldInputFieldProps, {}> {
```

Comment:
This component should be completely backwards compatible, but using the specific fields would provide better typing and flexibility